gltf/src/net/mgsx/gltf/scene3d/shaders/PBRShader.java

```diff
@@ -803,10 +826,13 @@ protected void bindLights(Renderable renderable, Attributes attributes) {
 				float clip = 3.f / (2 * mapSize);
 				
 				int unit = context.textureBinder.bind(light.getDepthMap());
-				program.setUniformi(u_csmSamplers + i, unit);
-				program.setUniformMatrix(u_csmTransforms + i, light.getProjViewTrans());
-				program.setUniformf(u_csmPCFClip + i, pcf, clip);
+				program.setUniformi(u_csmSamplers[i], unit);
+				System.arraycopy(light.getProjViewTrans().val, 0, csmTransforms, i*16, 16);
+				csmPCFClip[i*2] = pcf;
+				csmPCFClip[i*2+1] = clip;
 			}
+			program.setUniformMatrix4fv(u_csmTransforms, csmTransforms, 0, csmTransforms.length);
```

Comment:
should be cascadeCount * 16. Same for next line, should be cascadeCount * 2.

Another solution (a better one) is to initalize arrays based on cascadeCount inside constructor. This way we could remove MAX_CSM limit.

Reply:
You are right, I'll make an update.